cvexporter/quay_images.py

```diff
@@ -0,0 +1,151 @@
+import os
+import requests
+import logging
+from functools import lru_cache as lc
+
+
+QUAY_API = str(os.getenv("QUAY_API", "https://quay.io/api/v1/"))
+QUAY_TOKEN = os.getenv("QUAY_TOKEN")
+
+
+def severities():
+    return ('unknown',
+            'negligible',
+            'low',
+            'medium',
+            'high',
+            'critical',
+            'defcon1')
+
+
+def is_scanned(vuln_data):
+    if vuln_data.get('status') is None:
+        return False
+    if vuln_data.get('status') == "scanned":
+        return True
+    else:
+        return False
+
+
+def has_results(vuln_data):
+    if vuln_data['data']['Layer'].get('Features') is None:
+        return False
+    else:
+        return True
+
+
+def from_quay(imageurl):
+    return imageurl.startswith('docker-pullable://quay.io')
+
+
+def imageurl_parse(imageurl, item):
+    try:
+        image_uri_list = str(imageurl.split('@')[0]).split('/')
+        image_name = str(''.join(image_uri_list[-1]))  # noqa: F841
+        image_org = str(''.join(image_uri_list[-2]))  # noqa: F841
+        image_host = str(''.join(image_uri_list[-3]))  # noqa: F841
+        image_digesthash = str(imageurl.split('@')[-1])  # noqa: F841
+    except ValueError:
+        raise ValueError("Error parsing this image URL: {0}".format(imageurl))
+    return eval('image_' + item)
+
+
+def image_name(imageurl):
+    return imageurl_parse(imageurl, 'name')
+
+
+def image_org(imageurl):
+    return imageurl_parse(imageurl, 'org')
+
+
+def image_digesthash(imageurl):
+    return imageurl_parse(imageurl, 'digesthash')
+
+
+def image_digest(imageurl):
+    return str(imageurl_parse(imageurl, 'digesthash').split(':')[0])
+
+
+def image_hash(imageurl):
+    return str(imageurl_parse(imageurl, 'digesthash').split(':')[1])
+
+
+def image_host(imageurl):
+    return imageurl_parse(imageurl, 'host')
+
+
+def gen_secapi_url(imageurl, api_url=QUAY_API):
+    return str("{0}repository/{1}/{2}/manifest/{3}/security".format(QUAY_API,
+               image_org(imageurl),
```

Comment:
image['org']

Reply:
todo -- will need to re-write a few lines